pandas/io/common.py

```diff
@@ -186,7 +190,12 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
     ----------
     filepath_or_buffer : a url, filepath (str, py.path.local or pathlib.Path),
                          or buffer
+            supports 'https://username:password@fqdn.com:port/aaa.csv'
     encoding : the encoding to use to decode py3 bytes, default is 'utf-8'
+    compression:
+    auth: (str,str), default None. (username, password) for HTTP(s) basic auth
+    verify_ssl: Default True. If False, allow self signed and invalid SSL
+                 certificates for https
 
```

Comment:
Actually the `compression` already existed, but was not in the comments. I simply added it - to comments and left it empty because I was not too familiar to add best docs. I'll fix the rest